code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java

```diff
@@ -470,4 +471,11 @@ public void doWithLockNoResult() {
             }
         });
     }
+
+    @Override
+    public void setToken(Service service) {
+        String token = ApiKeyFilter.generateKeys()[1];
+        DataAccessor.fields(service).withKey(ServiceDiscoveryConstants.FIELD_TOKEN).set(token);
+        objectManager.persist(service);
```

Comment:
@ibuildthecloud I tried using objectManager.setFields(), but it does not set the token in the db.
objectManager.setFields(service, ServiceDiscoveryConstants.FIELD_TOKEN, token);
return objectManager.reload(service);

but this does not set the token in the db inside data[]
If I define the field in service.json and invoke the above, I get "Database record has changed" exception